### PR TITLE
Make kombu FIPS compatible

### PR DESCRIPTION
**Pull request by dralley**
Pull in changes made by upstream PR:
https://github.com/celery/kombu/pull/711

closes #3641
https://pulp.plan.io/issues/3641

### CODE REVIEW

**Review by daviddavis — APPROVED** [review]

File: packages/python-kombu/python-kombu.spec [hunk 1]
```diff
@@ -14,7 +14,9 @@ Source0:        https://github.com/celery/kombu/archive/v%{version}.tar.gz#/%{sr
 # This can be removed in 4.0.3+
 Patch0: qpid-transport-works-with-celery-4.patch
 Patch1: 785.patch
- 
+# This can be removed in 4.2+
```
Comment (ekohl) [inline packages/python-kombu/python-kombu.spec:17]: https://github.com/celery/kombu/commit/4e28c0b839171ffddc8cbb43e0909adcd25f7e75 is already included in 4.1.0

File: packages/python-kombu/711.patch [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+ 
+ def oid_from(instance, threads=True):
+-- 
+2.17.0
+
+
+From a8753637306fdeea655569935b88d1e7bc0bf260 Mon Sep 17 00:00:00 2001
+From: George Psarakis <giwrgos.psarakis@gmail.com>
+Date: Wed, 26 Apr 2017 21:32:01 +0300
+Subject: [PATCH 2/3] Test cases for generate_oid
```
Comment (ekohl) [inline packages/python-kombu/711.patch:46]: Technically this patch isn't needed

File: packages/python-kombu/711.patch [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+     connection = Mock()
+     connection.channel_errors = (KeyError,)
+-- 
+2.17.0
+
+
+From 259ac2e2bfc174ecd6d6f9e7cd31269a6166d337 Mon Sep 17 00:00:00 2001
+From: George Psarakis <giwrgos.psarakis@gmail.com>
+Date: Wed, 26 Apr 2017 21:40:01 +0300
+Subject: [PATCH 3/3] Add missing function import
```
Comment (ekohl) [inline packages/python-kombu/711.patch:89]: Technically this patch isn't needed

**Comment by pcreech** [comment]
This built successfully in koji

http://koji.katello.org/koji/taskinfo?taskID=99018

Merging, will build release build shortly

File: packages/python-kombu/python-kombu.spec (hunk 1 above)
Comment (dralley) [reply packages/python-kombu/python-kombu.spec:17]: The comment isn't strictly correct, but the sources in this package are for 4.0.2.  We're likely going to skip over 4.1

**Comment by pcreech** [comment]
http://koji.katello.org/koji/taskinfo?taskID=99030

The release number got modified a bit due to some cherry-pick conflicts (i didn't bring in fixes from the 2.16 branch to master before).
But we have a successful build now!

**Pull request merged**